src/Microsoft.Dnx.ApplicationHost/Program.cs

```diff
@@ -259,7 +247,33 @@ private Task<int> ExecuteMain(DefaultHost host, string applicationName, string[]
                 return Task.FromResult(1);
             }
 
-            return EntryPointExecutor.Execute(assembly, args, host.ServiceProvider);
+            return EntryPointExecutor.Execute(assembly, args, host.ServiceProvider)
+                        .ContinueWith(t =>
+                        {
+                            t.Exception?.Handle(e =>
```

Comment:
nice! ...but make sure that it does not convert non-compilation exceptions to `AggregateExceptions`.... (i.e. before if an application threw it would just report the actual exception and not the exception wrapped in `AggregateException`)